docs/arcadia-users-group/20240206-intro-to-formatting-and-linting/lesson.md

```diff
@@ -0,0 +1,425 @@
+
+#### Comments
+Comments are human-readable lines of text that can appear anywhere in the source code. In Python, they are denoted by the `#` character. Although they are ignored by the Python interpreter and might seem like an area where "anything goes", it is important to adhere to strict standards of grammar and style when writing comments, since they are often the only way to explain the purpose of a particular line or block of code.
+
+In particular, comments should be complete sentences with proper punctuation. They should be written in full English sentences and should be grammatically correct. Importantly, they should end with periods; this is not only grammatically correct but is also the only way to indicate to the reader that the comment is complete and was not accidentally truncated (or never completely written in the first place). 
+
+Most importantly, comments should generally be used to explain *why* a particular line or block of code is doing what it is doing, not *what* it is doing or *how* it is doing it (usually, this should be apparent from the code itself). 
+
+Here is an example of a good comment that explains *why* a CSV file is loaded with a particular set of parameters:
+```python
+# We can assume missing values are always represented by the string 'NaN' in the CSV file,
+# so the parameter 'na_values' is set to 'NaN'.
+df = pd.read_csv('data.csv', na_values='NaN', keep_default_na=False)
+```
+
+Here is another example of a good comment that explains a non-obvious implementation detail when computing the nth Fibonacci number:
+```python
+def calculate_fibonacci(n):
+    """
+    Calculate the nth Fibonacci number using naive recursion.
+    """
+    # By definition, the Fibonacci sequence starts with zero and one.
+    if n <= 0:
+        return 0
+    elif n == 1:
+        return 1
+    else:
+        return fibonacci(n - 1) + fibonacci(n - 2)
+```
+
+#### Things comments should *not* be used for
+There are two major ways in which comments are commonly misused. Note that avoiding these misuses is not specific to Python but is a general best practice that applies to all programming languages.
+
+1. As we alluded to above, comments should not be redundant with the code. They should not simply restate what the code is doing and they should not include information that is apparent or readily inferred from the code itself. Here are some examples of such redundant comments:
+    ```python
+    # find the minimum value and clamp it to 0.
+    min_value = max(min(values), 0)
+
+    # get all the .txt files in the input directory.
+    filepaths = input_dirpath.glob('*.txt')
+
+    # create the directory if it doesn't already exist.
+    if not os.path.exists(dirpath):
+        os.mkdir(dirpath)
+    ```
```

Comment:
this block doesn't format correctly when i viewed the page with `mkdocs serve`. I think it needs to be not indented? I think it will also mess up your numbering scheme potentially, but im not sure. 

I tried to make the suggested change but the triple backticks messed it up and wouldn't let me :(

Reply:
oof that's annoying. I'll just rewrite this to eliminate the need to use numbering (and therefore the indents).